File: packages/tutoriales/src/centro/textos.ts
import { InjectionToken } from '@angular/core'

/**
 * **Cómo se presenta el centro en cada producto.** Es lo único del centro que cambia
 * entre el backoffice y el sitio público: el resto de los textos —«Comenzar»,
 * «Pendiente», «Antes conviene hacer»— dicen lo mismo en los dos.
 *
 * Sin esto, el sitio público le decía a un visitante sin cuenta que estaba aprendiendo
 * a usar «el backoffice».
 */
export interface PresentacionDelCentro {
  readonly titulo: string
  readonly proposito: string
}

export const PRESENTACION_DEL_CENTRO = new InjectionToken<PresentacionDelCentro>('PresentacionDelCentro')

/** Lo que se muestra si nadie dice otra cosa. */
export const presentacionPorOmision: PresentacionDelCentro = {
  titulo: 'Centro de tutoriales',
  proposito: 'Aprender haciendo, sobre las pantallas de verdad. Ningún tutorial toca datos ni confirma operaciones.',
}

/** Los textos del centro de tutoriales, en voz de marca. Un archivo por dominio. */
export const textosAyuda = {
  buscar: 'Buscar un tutorial',
  filtrarEstado: 'Filtrar por estado',
  filtrarCategoria: 'Filtrar por módulo',
  todos: 'Todos',
  pendientes: 'Pendientes',
  enProgreso: 'En progreso',
  completados: 'Completados',
  obligatorios: 'Obligatorios',
  recomendado: 'Recomendado para vos',
  avanceTitulo: 'Tu avance',
  avanceEtiqueta: 'Avance general de los tutoriales',
  avanceDe: (hechos: number, total: number): string => `${hechos} de ${total} tutoriales completados`,
  sinResultados: 'Ningún tutorial coincide con lo que buscaste. Probá con otra palabra o sacá los filtros.',
  sinTutoriales: 'Todavía no hay tutoriales para tu rol. Cuando los haya, aparecen acá.',
  comenzar: 'Comenzar',
  continuar: 'Continuar',
  repetir: 'Repetir',
  reiniciar: 'Reiniciar',
  minutos: (n: number): string => `${n} min`,
  pasos: (n: number): string => (n === 1 ? '1 paso' : `${n} pasos`),
  obligatorio: 'Obligatorio',
  requisitos: 'Antes conviene hacer',
  estado: {
    pendiente: 'Pendiente',
    'en-progreso': 'En progreso',
    completado: 'Completado',
    omitido: 'A medias',
  },
  dificultad: {
    inicial: 'Inicial',
    intermedio: 'Intermedio',
    avanzado: 'Avanzado',
  },
  problemasTitulo: 'Problemas en el catálogo',
  problemasAyuda: 'Esto lo ve cualquiera que entre, a propósito: un tutorial mal escrito es un error de la herramienta, no del catálogo secreto de alguien.',
} as const

File: apps/movil/lib/pantallas/soporte/textos.dart
/// Los textos del centro de ayuda, en voz de marca. Un archivo por dominio.
abstract final class TextosSoporte {
  static const titulo = 'Ayuda';
  static const proposito =
      'Aprendé a usar AportaYa haciéndolo, sobre las pantallas de verdad. Ningún tutorial mueve tu plata ni confirma nada por vos.';
  static const buscar = 'Buscar un tutorial';
  static const todos = 'Todos';
  static const pendientes = 'Pendientes';
  static const enProgreso = 'En progreso';
  static const completados = 'Completados';
  static const avanceTitulo = 'Tu avance';
  static const avanceEtiqueta = 'Avance general de los tutoriales';
  static const sinResultados =
      'Ningún tutorial coincide con lo que buscaste. Probá con otra palabra.';
  static const sinTutoriales =
      'Todavía no hay tutoriales para vos. Cuando los haya, aparecen acá.';
  static const recomendado = 'Te conviene empezar por acá';
  static const comenzar = 'Comenzar';
  static const continuar = 'Continuar';
  static const repetir = 'Repetir';
  static const reiniciar = 'Reiniciar';
  static const obligatorio = 'Importante';
  static const requisitos = 'Antes conviene hacer';
  static const siguiente = 'Siguiente';
  static const terminar = 'Terminar';
  static const omitir = 'Dejar el tutorial';
  static const salirTitulo = '¿Dejamos el tutorial acá?';
  static const salirCuerpo =
      'Guardamos por qué paso ibas. Podés retomarlo cuando quieras desde Ayuda.';
  static const salirConfirmar = 'Dejarlo por ahora';
  static const salirSeguir = 'Seguir';
  static const comoFunciona = 'Cómo funciona esta pantalla';
  static const problemasTitulo = 'Hay tutoriales mal configurados';

  static String avanceDe(int hechos, int total) =>
      '$hechos de $total tutoriales completados';
  static String minutos(int n) => '$n min';
  static String pasos(int n) => n == 1 ? '1 paso' : '$n pasos';
}
